rasa/shared/nlu/training_data/training_data.py

```diff
@@ -79,6 +79,14 @@ def fingerprint(self) -> Text:
 
         return rasa.shared.utils.io.deep_container_fingerprint(relevant_attributes)
 
+    def label_fingerprint(self) -> Text:
+        """Fingerprint the labels in the training data.
+
+        Returns:
+            hex string as a fingerprint of the training data labels.
+        """
+        return rasa.shared.utils.io.deep_container_fingerprint(self.intents)
```

Comment:
In my opinion this should include entity names as well

Reply:
It should include `entities`, `entity_roles` and `entity_groups`.